conformance/data_model/decimal.ion

```diff
@@ -0,0 +1,458 @@
+               (binary "73 FB 46 0C")
+               (denotes (Decimal 3142 -3)))
+         (each (text "3.14159")
+               (binary "74 F7 2F CB 04")
+               (denotes (Decimal 314159 -5)))
+         (each (text "3.1415927")
+               (binary "75 F3 77 5E DF 01")
+               (denotes (Decimal 31415927 -7)))
+         (each (text "3.141592653")
+               (binary "76 EF 4D E6 40 BB 00")
+               (denotes (Decimal 3141592653 -9)))
+         (each (text "3.141592653590")
+               (binary "77 E9 16 9F 83 75 DB 02")
+               (denotes (Decimal 3141592653590 -12)))
+         (each (text "3.14159265358979323")
+               (binary "79 DF FB A0 9E F6 2F 1E 5C 04")
+               (denotes (Decimal 314159265358979323 -17)))
+         (each (text "3.1415926535897932384626")
+               (binary "7B D5 72 49 64 CC AF EF 8F 0F A7 06")
+               (denotes (Decimal 31415926535897932384626 -22)))
+         (each (text "3.141592653589793238462643383")
+               (binary "7D CB B7 3C 92 86 40 9F 1B 01 1F AA 26 0A")
+               (denotes (Decimal 3141592653589793238462643383 -27)))
+         (each (text "3.14159265358979323846264338327950")
+               (binary "7F C1 8E 29 E5 E3 56 D5 DF C5 10 8F 55 3F 7D 0F")
+               (denotes (Decimal 314159265358979323846264338327950 -32)))
+         (each (text "3.141592653589793238462643383279503")
+               (binary "F7 21 BF 8F 9F F3 E6 64 55 BE BA A7 96 57 79 E4 9A 00")
+               (denotes (Decimal 3141592653589793238462643383279503 -33))))
+
+(ion_1_x "in Ion text, a decimal with unnecessary leading zeros before the decimal point is invalid"
+         (each (text "00.")
+               (text "-00.")
+               (text "053.")
+               (text "00000053.")
+               (text "-054.")
+               (text "-00000054.")
+               (signals "invalid leading zero")))
+
+(ion_1_x "in Ion text, a decimal with no digits before the decimal point is invalid"
+         (each (text ".0")
+               (text ".1d2")
+               (text "-.01")
+               (signals "syntax error")))
+
+(ion_1_x "a decimal with an unexpected non-numeric character should raise an error"
+         (each "an unexpected non-numeric character"
+               (text "1.a")
+               (text "0x21.34")
+               (text "0x1g")
+               (text "0.1d-0x2A")
+               (text "0c10")
+               (text "0b1a")
+               "spacing underscores between characters that are not both base 10 digits"
+               (text "1_.2d3")
+               (text "1._2d3")
+               (text "1.2_d3")
+               (text "1.2d_3")
+               (text "1.2d3_")
+               "consecutive underscores"
+               (text "-6__0.2")
+               (signals "numeric value followed by invalid character")))
+
+// TODO: Should we test that incomplete binary decimals signal an error? (i.e. EOF is reached before the end of the value.)
```

Comment:
I would love to have a way to generally test "incomplete data" behavior, but it is closely tied to configuration options, so I'm not sure if the conformance tests are the right place. In ion-java, we have "non-continuable" mode, which fails at the first sign of incomplete data, and "continuable" mode, which can yield back to the caller and check later for more data. Maybe we can have opt-in conformance tests for implementations that support continuability? We don't need to solve this here.